### PR TITLE
Provide a client for Azure OpenAI GPT model service

### PR DESCRIPTION
**Pull request by hydraxman**
**Linked GitHub issue ID**: #  

## Pull Request Checklist

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Code compiles correctly with all tests are passed.
- [x] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [x] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [ ] Yes
- [x] No

## Pull Request Description

**A few words to explain your changes**:  Provide a client for Azure OpenAI GPT model service. Only add the code, no function attached.

### How you tested it
*Please make sure the change is tested, you can test it by adding UTs, do local test and share the screenshots, etc.*

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Technical design
- [ ] Build related changes
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe): 

### Feature UI screenshots or Technical design diagrams
*If this is a relatively large or complex change, kick it off by drawing the tech design with PlantUML and explaining why you chose the solution you did and what alternatives you considered, etc...*

### CODE REVIEW

**Comment by hydraxman** [comment]
@zhou9584 this CI build keeps failing in the run UT task, and I can pass it locally, could you help me understand why?

File: center/src/main/java/com/microsoft/hydralab/center/openai/ChatRequest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+public class ChatRequest {
+    private List<ChatMessage> messages;
+    @JSONField(name = "max_tokens")
+    private int maxTokens = 800;
+    private double temperature = 0.7;
```
Comment (hydraxman) [inline center/src/main/java/com/microsoft/hydralab/center/openai/ChatRequest.java:13]:
```suggestion
    private double temperature = 0.75;
```

File: center/src/main/java/com/microsoft/hydralab/center/openai/AzureOpenAIServiceClient.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+public class AzureOpenAIServiceClient {
```
Comment (hydraxman) [inline center/src/main/java/com/microsoft/hydralab/center/openai/AzureOpenAIServiceClient.java:14]:
```suggestion
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
public class AzureOpenAIServiceClient {
```

File: center/src/main/java/com/microsoft/hydralab/center/openai/ChatMessage.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.microsoft.hydralab.center.openai;
+
+import lombok.Data;
+
+@Data
```
Comment (hydraxman) [inline center/src/main/java/com/microsoft/hydralab/center/openai/ChatMessage.java:5]:
```suggestion
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
@Data
```

File: center/src/main/java/com/microsoft/hydralab/center/openai/ChatRequest.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.microsoft.hydralab.center.openai;
+
+import com.alibaba.fastjson.annotation.JSONField;
+import lombok.Data;
+
+import java.util.List;
+
+@Data
```
Comment (hydraxman) [inline center/src/main/java/com/microsoft/hydralab/center/openai/ChatRequest.java:8]:
```suggestion
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
@Data
```

File: center/src/test/java/com/microsoft/hydralab/center/openai/AzureOpenAIServiceClientTest.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.microsoft.hydralab.center.test.BaseTest;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.util.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.Arrays;
+
+@ActiveProfiles("local")
```
Comment (hydraxman) [inline center/src/test/java/com/microsoft/hydralab/center/openai/AzureOpenAIServiceClientTest.java:11]:
```suggestion
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
@ActiveProfiles("local")
```

File: center/src/main/java/com/microsoft/hydralab/center/openai/AzureOpenAIServiceClient.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    public String chatCompletion(ChatRequest request) {
+        MediaType mediaType = MediaType.parse("application/json");
+        String url = String.format("%s/openai/deployments/%s/chat/completions?api-version=%s", endpoint, deployment, apiVersion);
+
+        String requestBodyString = JSON.toJSONString(request);
+        logger.info("Request body: {}", requestBodyString);
+
+        RequestBody body = RequestBody.create(requestBodyString, mediaType);
+        Request httpRequest = new Request.Builder().url(url).post(body)
+//                .addHeader("Content-Type", "application/json")
```
Comment (hydraxman) [inline center/src/main/java/com/microsoft/hydralab/center/openai/AzureOpenAIServiceClient.java:40]:
```suggestion
```

**Review by DexterDreeeam — APPROVED** [review]

**Pull request merged**